packages/firebase_storage/firebase_storage/windows/firebase_storage_plugin.cpp

```diff
@@ -0,0 +1,807 @@
+      GetCPPStorageFromPigeon(pigeonApp, pigeonReference.bucket());
+  return cpp_storage->GetReference(pigeonReference.full_path());
+}
+
+flutter::BinaryMessenger*
+    firebase_storage_windows::FirebaseStoragePlugin::messenger_ = nullptr;
+std::map<std::string,
+         std::unique_ptr<flutter::EventChannel<flutter::EncodableValue>>>
+    event_channels_;
+std::map<std::string, std::unique_ptr<flutter::StreamHandler<>>>
+    stream_handlers_;
+
+std::string RegisterEventChannelWithUUID(
+    std::string prefix, std::string uuid,
+    std::unique_ptr<flutter::StreamHandler<flutter::EncodableValue>> handler) {
+  std::string channelName = prefix + "/" + uuid;
+
+  event_channels_[channelName] =
+      std::make_unique<flutter::EventChannel<flutter::EncodableValue>>(
+          FirebaseStoragePlugin::messenger_, channelName,
+          &flutter::StandardMethodCodec::GetInstance());
+  stream_handlers_[channelName] = std::move(handler);
+
+  event_channels_[channelName]->SetStreamHandler(
+      std::move(stream_handlers_[channelName]));
+
+  return uuid;
+}
+
+std::string RegisterEventChannel(
+    std::string prefix,
+    std::unique_ptr<flutter::StreamHandler<EncodableValue>> handler) {
+  UUID uuid;
+  UuidCreate(&uuid);
+  char* str;
+  UuidToStringA(&uuid, (RPC_CSTR*)&str);
+
+  std::string channelName = prefix + "/" + str;
+
+  event_channels_[channelName] =
+      std::make_unique<flutter::EventChannel<flutter::EncodableValue>>(
+          FirebaseStoragePlugin::messenger_, channelName,
+          &flutter::StandardMethodCodec::GetInstance());
+  stream_handlers_[channelName] = std::move(handler);
+
+  event_channels_[channelName]->SetStreamHandler(
+      std::move(stream_handlers_[channelName]));
+
+  return str;
+}
+
+std::string FirebaseStoragePlugin::GetStorageErrorCode(Error storageError) {
+  switch (storageError) {
+    case firebase::storage::kErrorNone:
+      return "unknown";
+    case firebase::storage::kErrorUnknown:
+      return "unknown";
+    case firebase::storage::kErrorObjectNotFound:
+      return "object-not-found";
+    case firebase::storage::kErrorBucketNotFound:
+      return "bucket-not-found";
+    case firebase::storage::kErrorProjectNotFound:
+      return "project-not-found";
+    case firebase::storage::kErrorQuotaExceeded:
+      return "quota-exceeded";
+    case firebase::storage::kErrorUnauthenticated:
+      return "unauthenticated";
+    case firebase::storage::kErrorUnauthorized:
+      return "unauthorized";
+    case firebase::storage::kErrorRetryLimitExceeded:
+      return "retry-limit-exceeded";
+    case firebase::storage::kErrorNonMatchingChecksum:
+      return "invalid-checksum";
+    case firebase::storage::kErrorDownloadSizeExceeded:
+      return "download-size-exceeded";
+    case firebase::storage::kErrorCancelled:
+      return "canceled";
+
+    default:
+      return "unknown";
+  }
+}
+
+std::string FirebaseStoragePlugin::GetStorageErrorMessage(Error storageError) {
```

Comment:
For the error message, the future is supposed to contain the error message directly
`completed_future.error_message();`
Maybe it's not the case with Storage ?

Reply:
it is, however it is not exact same words with the other two platforms (iOS/Android), so do a conversion here for consistency.